java/java.hints/src/org/netbeans/modules/java/hints/TreeShims.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints;
+
+import com.sun.source.tree.CaseTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.Tree;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.List;
+
+public class TreeShims {
+
```

Comment:
Not "necessary", but felt wrong to plague the public API with utilities that we know won't be needed eventually. (And when we know that eventually many new methods likes this will be needed, leading to an ever growing class, where only a subset of the methods make sense, but can't be removed, because they are API.)

My long-term intent was to have only one copy, and copy it from java.source(.base) to java.hints at build time, although I didn't go that far in this patch.

I suspect this may not be the ideal model, but it allows us to continue, without committing to things to which we shouldn't commit.